data-plane/receiver/src/main/java/dev/knative/eventing/kafka/broker/receiver/impl/auth/OIDCDiscoveryConfigListener.java

```diff
@@ -36,24 +35,25 @@ public class OIDCDiscoveryConfigListener implements AutoCloseable {
     private final Vertx vertx;
     private final FileWatcher configFeaturesWatcher;
     private final int timeoutSeconds;
-    private List<Consumer<OIDCDiscoveryConfig>> callbacks;
-    private OIDCDiscoveryConfig oidcDiscoveryConfig;
+    private final CopyOnWriteArrayList<Consumer<OIDCDiscoveryConfig>> callbacks;
+    private final AtomicReference<OIDCDiscoveryConfig> oidcDiscoveryConfig;
 
     public OIDCDiscoveryConfigListener(String featuresConfigPath, Vertx vertx, int timeoutSeconds) throws IOException {
         this.featuresConfigPath = featuresConfigPath;
         this.vertx = vertx;
         this.timeoutSeconds = timeoutSeconds;
+        this.oidcDiscoveryConfig = new AtomicReference<>();
+        this.callbacks = new CopyOnWriteArrayList<>();
 
         this.buildFeaturesAndOIDCDiscoveryConfig();
 
         this.configFeaturesWatcher =
                 new FileWatcher(new File(featuresConfigPath + "/" + FeaturesConfig.KEY_AUTHENTICATION_OIDC), () -> {
-                    if (this.oidcDiscoveryConfig == null) {
+                    if (this.oidcDiscoveryConfig.get() == null) {
                         this.buildFeaturesAndOIDCDiscoveryConfig();
-                        if (this.oidcDiscoveryConfig != null && this.callbacks != null) {
-                            this.callbacks.stream()
-                                    .filter(Objects::nonNull)
-                                    .forEach(c -> c.accept(this.oidcDiscoveryConfig));
+                        OIDCDiscoveryConfig config = this.oidcDiscoveryConfig.get();
+                        if (config != null) {
+                            this.callbacks.forEach(callback -> callback.accept(config));
```

Comment:
I see in `deregisterCallback` we set `callback` (the element) to null, don't we need to check that `callback` is not null before calling `accept` ?